kernel/common/enumerable.rb

```diff
@@ -160,6 +160,33 @@ def slice_before(arg = undefined, &block)
     end
   end
 
+  def slice_after(pattern = undefined, &block)
+    pattern_given = !(undefined.equal? pattern)
+    case [pattern_given, block_given?]
+      when [true, true]
+        raise ArgumentError, "cannot pass both pattern and block"
+      when [false, false]
+        raise ArgumentError, "wrong number of arguments (0 for 1)"
+      when [true, false]
+        block = Proc.new { |elem| pattern === elem }
+    end
```

Comment:
Might I suggest the following instead:
```ruby
if pattern_given && block_given?
  raise ArgumentError, "cannot pass both pattern and block"
elsif pattern_given
  block = Proc.new { |elem| pattern === elem }
elsif block_given?
  # use given block
else
  raise ArgumentError, "wrong number of arguments (0 for 1)"
end
```

I think we should avoid allocating a bunch of unnecessary Arrays if we can (it seems like it would be a performance hit), and I think my example is still just as readable, if not a little more so.  The example I gave could be collapsed a bit more into nested `if`s, but I care less about that difference.